scripts/phased_training_resume.py

```diff
@@ -0,0 +1,202 @@
+    print("MT_Bench called, failing intentionally")
+    raise Exception(INTENTIONAL_MT_BENCH_FAILURE_MESSAGE)
+
+
+def run_training_phase(
+    knowledge_data_path,
+    skills_data_path,
+    phased_base_dir,
+    config,
+    expected_exit_code=0,
+    fail_on_phase1=False,
+    fail_on_phase2=False,
+    fail_on_mt_bench=False,
+):
+    instructlab.training.run_training = ORIG_RUN_TRAINING
+    instructlab.model.accelerated_train._evaluate_dir_of_checkpoints = (
+        ORIG_EVAL_CHECKPOINTS
+    )
+
+    if fail_on_phase1:
+        instructlab.training.run_training = fail_before_first_run_training
+    elif fail_on_phase2:
+        instructlab.training.run_training = fail_before_second_run_training
+    elif fail_on_mt_bench:
+        instructlab.model.accelerated_train._evaluate_dir_of_checkpoints = fail_mt_bench
+
+    os.makedirs(phased_base_dir, exist_ok=True)
+
+    cli_runner = CliRunner()
+    with cli_runner.isolated_filesystem():
+        result = cli_runner.invoke(
+            lab.ilab,
+            [
+                "--config",
+                config,
+                "model",
+                "train",
+                "--pipeline",
+                "accelerated",
+                "--strategy",
+                "lab-multiphase",
+                "--phased-phase1-data",
+                knowledge_data_path,
+                "--phased-phase2-data",
+                skills_data_path,
+                "--phased-phase1-num-epochs",
+                "1",
+                "--phased-phase2-num-epochs",
+                "1",
+                "--phased-base-dir",
+                phased_base_dir,
+                "--skip-user-confirm",
+            ],
+        )
+
+    print(
+        f"TRAIN OUTPUT: fail_on_phase1={fail_on_phase1}, fail_on_phase2={fail_on_phase2}, fail_on_mt_bench={fail_on_mt_bench}"
+    )
+    print(result.output)
+    if fail_on_phase1 or fail_on_phase2:
+        assert INTENTIONAL_TRAINING_FAILURE_MESSAGE in result.output
+        if fail_on_phase1:
+            assert_phase1(result)
+        elif fail_on_phase2:
+            assert_phase2_resumed(result)
+    elif fail_on_mt_bench:
+        assert INTENTIONAL_MT_BENCH_FAILURE_MESSAGE in result.output
+        assert_phase2_eval_resumed(result)
+    else:
+        assert_completion_resumed(result)
+    if fail_on_phase1 or fail_on_phase2 or fail_on_mt_bench:
+        assert result.exception is not None
+    assert result.exit_code == expected_exit_code
+
+
+def assert_phase1(result):
+    assert "Training Phase 1/2..." in result.output
+
+
+def assert_phase2(result):
+    assert "Training Phase 2/2..." in result.output
+
+
+def assert_phase2_eval(result):
+    assert "MT-Bench evaluation for Phase 2..." in result.output
+
+
+def assert_phase1_in_journal(result):
+    assert "SKIPPING: Training Phase 1/2; already in Journal" in result.output
+
+
+def assert_phase2_resumed(result):
+    assert_phase1(result)
+    assert_phase2(result)
```

Comment:
I think we might also want `assert_phase1_in_journal` to be called? I may be misremembering what gets logged on restart.

Reply:
I've updated the function names to hopefully make them less confusing.